week2/homework/src/index.js

```diff
@@ -1,3 +1,115 @@
 'use strict';
+const fs = require('fs');
+const commandOption = process.argv[2];
+const itemIndex = process.argv[3] - 1;
+const myArguments = process.argv.slice(3);
+const uniCode = 'utf8';
+const myCommandFile = 'command.txt';
 
-// TODO: Write the homework code in this file
+function printHelp() {
+  console.log(`Usage: node index.js [options]
+
+HackYourFuture Node.js Week 2 - Lecture To-Do App
+
+Options:
+
+  read          read all to-dos
+  write [to-do] add to-do
+  help          show this help text
+  `);
+}
+
+function showCommandList() {
+    return new Promise ((resolve, reject) => {
+        fs.readFile(myCommandFile, uniCode, (error, data) => {
+            error ? reject(error) : resolve(data);
+        });
+    });
+}
+
+function addCommand(...args) {
+    return new Promise((resolve, reject) => {
+        fs.appendFile(myCommandFile, `${args.join(' ')}\n`, (error, data) => {
+            error ? reject(error) : resolve(data);
+        });
+    });
+}
+
+function removeOrUpdateCommand(dataIndex, cmd, ...args) {
+    let fileFields = [];
+    if(cmd === 'remove') {
+        fs.readFile(myCommandFile, uniCode, (error, data) => {
+            if(error) console.error;
+            else {
+                fileFields = data.split('\n');
+                fileFields.splice(dataIndex, 1);
+                let temp = ' ';
+                fileFields.forEach((fileField) => {
+                    temp += fileField + '\n';
+                });
+                console.log('Temp array : ', temp);
+                console.log(fileFields);
+                fs.writeFile(myCommandFile, temp.trim() + '\n', uniCode, (err) => {
+                    if(err) return err;
+                });
+            }
+        });
+    } else if (cmd === 'update') {
```

Comment:
since `update` and `remove` are two completely different blocks, this should be two different functions.